src/openforms/submissions/tests/form_logic/test_binding.py

```diff
@@ -326,6 +326,60 @@ def test_it_sends_the_body_as_json(self, m):
         self.assertIn(("Content-Type", "application/json"), request.headers.items())
         self.assertEqual(request.body, b'"Armour"')
 
+    @requests_mock.Mocker()
+    def test_it_sends_the_body_as_json_with_variables(self, m):
+        m.get("https://httpbin.org/anything", json={"foo": "bar"})
+
+        var = FormVariableFactory.build(
+            service_fetch_configuration=ServiceFetchConfigurationFactory.build(
+                service=self.service,
+                path="anything",
+                body={"foo": "{{some_variable}}"},
+            )
+        )
+
+        _ = bind(var, {"some_variable": "bar"})
+        request = m.last_request
+
+        self.assertIn(("Content-Type", "application/json"), request.headers.items())
+        self.assertEqual(request.body, b'{"foo": "bar"}')
+
+    @requests_mock.Mocker()
+    def test_it_sends_the_body_as_json_with_variables_nested(self, m):
+        m.get("https://httpbin.org/anything", json={"foo": {"nested": "bar"}})
+
+        var = FormVariableFactory.build(
+            service_fetch_configuration=ServiceFetchConfigurationFactory.build(
+                service=self.service,
+                path="anything",
+                body={"foo": {"nested": "{{some_variable}}"}},
+            )
+        )
+
+        _ = bind(var, {"some_variable": "bar"})
+        request = m.last_request
+
+        self.assertIn(("Content-Type", "application/json"), request.headers.items())
+        self.assertEqual(request.body, b'{"foo": {"nested": "bar"}}')
+
+    @requests_mock.Mocker()
+    def test_it_sends_the_body_as_json_with_variables_escape_quotes(self, m):
+        m.get("https://httpbin.org/anything", json={"foo": "%22bar"})
+
+        var = FormVariableFactory.build(
+            service_fetch_configuration=ServiceFetchConfigurationFactory.build(
+                service=self.service,
+                path="anything",
+                body={"foo": "{{some_variable}}"},
+            )
+        )
+
+        _ = bind(var, {"some_variable": '"bar'})
```

Comment:
I'm not sure if there are other examples of values that could potentially break the JSON body